File: firmware_factory_control_unit/builder/src/board-archive.ts
// Pfade ins Board-Archiv (s. docs/build-process.md Abschnitt 3/5): ein Ordner pro physischem
// Board unter boardsDir(), darin "generated/" als vollstaendiger, bei jedem Lauf ueberschriebener
// Snapshot aller fuer dieses Board zuletzt erzeugten Artefakte.
import path from "node:path";
import { boardsDir } from "./environment-config.ts";

export function boardArchiveDir(boardId: string): string {
	return path.join(boardsDir(), boardId);
}

export function boardGeneratedDir(boardId: string): string {
	return path.join(boardArchiveDir(boardId), "generated");
}

File: firmware_factory_control_unit/builder/BoardArchive.cs
// Pfade ins Board-Archiv (s. docs/build-process.md Abschnitt 3/5).
namespace Builder;

public static class BoardArchive
{
	public static string BoardArchiveDir(string boardId) => Path.Combine(BuilderOptions.Current.BoardsDir, boardId);

	public static string BoardGeneratedDir(string boardId) => Path.Combine(BoardArchiveDir(boardId), "generated");
}
